activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/process/ProcessModelValidator.java

```diff
@@ -42,49 +38,62 @@
  */
 public class ProcessModelValidator implements ModelContentValidator {
 
-    private final Logger log = LoggerFactory.getLogger(ProcessModelValidator.class);
+  private final Logger log = LoggerFactory.getLogger(ProcessModelValidator.class);
 
-    private final ProcessModelType processModelType;
+  private final ProcessModelType processModelType;
 
-    private final Set<BpmnCommonModelValidator> bpmnCommonModelValidators;
+  private final Set<BpmnCommonModelValidator> bpmnCommonModelValidators;
 
-    private final ProcessModelContentConverter processModelContentConverter;
+  private final ProcessModelContentConverter processModelContentConverter;
 
-    public ProcessModelValidator(ProcessModelType processModelType,
-                                 Set<BpmnCommonModelValidator> bpmnCommonModelValidators,
-                                 ProcessModelContentConverter processModelContentConverter) {
-        this.processModelType = processModelType;
-        this.bpmnCommonModelValidators = bpmnCommonModelValidators;
-        this.processModelContentConverter = processModelContentConverter;
-    }
-
-    @Override
-    public void validate(byte[] bytes,
-        ValidationContext validationContext) {
+  public ProcessModelValidator(ProcessModelType processModelType,
+                               Set<BpmnCommonModelValidator> bpmnCommonModelValidators,
+                               ProcessModelContentConverter processModelContentConverter) {
+    this.processModelType = processModelType;
+    this.bpmnCommonModelValidators = bpmnCommonModelValidators;
+    this.processModelContentConverter = processModelContentConverter;
+  }
 
-        BpmnModel bpmnModel = processModelContentConverter.convertToBpmnModel(bytes);
+  @Override
+  public void validate(byte[] bytes,
+                       ValidationContext validationContext) {
 
-        List<ModelValidationError> validationErrors = bpmnCommonModelValidators
-            .stream()
-            .flatMap(bpmnCommonModelValidator -> bpmnCommonModelValidator.validate(bpmnModel,
-                validationContext))
-            .collect(Collectors.toList());
+    List<ModelValidationError> validationErrors = getModelValidationErrors(bytes,
+                                                                           validationContext);
 
-        if (!validationErrors.isEmpty()) {
-            String messageError = "Semantic process model validation errors encountered: " + validationErrors;
-            log.debug(messageError);
-            throw new SemanticModelValidationException(messageError,
-                                                       validationErrors);
-        }
+    if (!validationErrors.isEmpty()) {
```

Comment:
See the comment on `ModelValidationError` as root for all the errors.
If we decide to keep it as root for error hierarchy, we should probably change the signature of `BpmnCommonModelValidator` since it could lead to unwanted implementations.